qa/os/src/test/java/org/elasticsearch/packaging/util/Docker.java

```diff
@@ -281,6 +284,76 @@ public static boolean existsInContainer(Path path) {
         return result.isSuccess();
     }
 
+    /**
+     * Run privilege escalated shell command on the local file system via a bind mount inside a Docker container.
+     * @param shellCmd The shell command to execute on the localPath e.g. `mkdir /containerPath/dir`.
+     * @param localPath The local path where shellCmd will be executed on (inside a container).
+     * @param containerPath The path to mount localPath inside the container.
+     */
+    private static void executePrivilegeEscalatedShellCmd(String shellCmd, Path localPath, Path containerPath) {
+        final List<String> args = new ArrayList<>();
+
+        args.add("docker run");
+
+        // Don't leave orphaned containers
+        args.add("--rm");
+
+        // Mount localPath to a known location inside the container, so that we can execute shell commands on it later
+        args.add("--volume \"" + localPath.getParent() + ":" + containerPath.getParent() + "\"");
+
+        // Use a lightweight musl libc based small image
+        args.add("alpine");
+
+        // And run inline commands via the POSIX shell
+        args.add("/bin/sh -c \"" + shellCmd + "\"");
+
+        final String command = String.join(" ", args);
+        logger.info("Running command: " + command);
+        sh.run(command);
+    }
+
+    /**
+     * Create a directory with specified uid/gid using Docker backed privilege escalation.
+     * @param localPath The path to the directory to create.
+     * @param uid The numeric id for localPath
+     * @param gid The numeric id for localPath
+     */
+    public static void mkDirWithPrivilegeEscalation(Path localPath, Integer uid, Integer gid) {
```

Comment:
Why `Integer` instead of `int`?

Reply:
This is because the result of `getNumericUnixPathOwnership` is stored in the value of a Map in https://github.com/elastic/elasticsearch/pull/49529/files#diff-4d0f4fd724af524a1c9c1bda1873cd0eR251

If you prefer I can also use switch the signature to use `int` here as you suggested and rely on `.intValue()` in the two assertions below.

Reply:
I implemented the latter approach described above in 17d22b7ce193849fe2bc2ef96e97109e4c668765